api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -1693,6 +1704,40 @@ public void getDefaultLocaleForUser_shouldReturnDefaultLocaleForUserIfAlreadySet
 		assertEquals(Locale.FRENCH, locale);
 	}
 
+	@Test
+	public void getLastLoginTimeForUser_shouldReturnEmptyStringOnLastLoginTimeIfPropertyNotSet() {
+		User createdUser = createTestUser();
+		assertEquals("", createdUser.getUserProperty(OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP));
+		assertEquals("", Context.getUserService().getLastLoginTime(createdUser));
+	}
+
+	@Test
+	public void getLastLoginTimeForUser_shouldReturnEmptyStringOnLastLoginTimeIfADifferentUserIsLoggedIn() {
+		executeDataSet(XML_FILENAME);
+		User createdUser = createTestUser();
+		Context.authenticate(getTestUserCredentials());
+		
+		assertEquals("", createdUser.getUserProperty(OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP));
+		assertEquals("", Context.getUserService().getLastLoginTime(createdUser));
+	}
+	
+	@Test
+	public void getLastLoginTimeForUser_shouldReturnLastLoginTimeStampIfPropertyIsSet() {
+		executeDataSet(XML_FILENAME);
+		User createdUser = createTestUser();
+		Context.authenticate(new UsernamePasswordCredentials("bwolfe", "Openmr5xy"));
+		
+		assertThat(createdUser.getUserProperty(OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP))
+			.isNotNull()
+			.isNotEmpty();
+		assertThat(Long.parseLong(Context.getUserService().getLastLoginTime(createdUser)))
```

Comment:
If this test is all about ensuring that the last login timestamp is set, as per the test name, shouldn't it all be about simply asserting that the return value is not empty?